### DISCUSSION TITLE
Clone / "Clone in place" for fields

### DISCUSSION THREAD

**Discussion by MattWilcox** [craftcms/cms#14566] [Ideas] [answered]
With the new content modelling it's being encouraged to use "generic" base fields, and to reuse them in multiple Entry Types etc, with the option of re-naming both the title and the handle where those field-instances are used.

This works well, until one specific instance use requires some customisation that isn't the handle or title - for example, if there's a light-switch field "show additional details" which is default disabled, but a specific instance use needs to be default enabled.

This requires making a new field at the moment. I'm finding that I'm creating fields "in place" quite often in 5, rather than in the Fields section.

It would be handy if there was a "Clone in place" option, that would have Craft create a clone of the field who's settings could be edited, and which would automatically de-assign the old field and assign the new one.

Templates will likely still need to be changed, but it'd be a neat enhancement.

**Comment by thupsi** [comment]
+1 from me!

**Comment by thupsi** [comment]
IMO this would be a great follow up to the `fields/merge` and `entry-types/merge` commands coming in 5.3 🙂 And of course, `ckeditor/convert/matrix`, which I hope will be in a CKEditor relase sooner rather than later!

**Reply by brandonkelly** [reply]
I actually made an attempt at this earlier this week, but ran into a couple snags. I do see the need for it though, and will give it another shot when I’m not quite as under the gun.

**Reply by mmikkel** [reply]
...although a UI implementation for this would be nice, I'd be 100% satisfied with something like a `fields/split` command which worked similarly to `fields/merge`, but in reverse. 

I.e. something like `craft fields/split bodyText` – then the command would 

- give a prompt for entering the new field's name and handle, 
- for each usage for the original `bodyText` field, prompt the user if they want to keep the original `bodyText` field for that usage, or migrate to the new field
- in the end, project config changes and content migrations are generated and can be deployed to production

**Reply by thupsi** [reply]
I agree that a cli command would be perfectly adequate and even more powerful for batch splitting many usages.

**Comment by ryanleichty** [comment]
+1 👍 Had a similar idea though slightly different: https://github.com/craftcms/cms/discussions/17893

**Comment by alexanderloewe** [comment]
+10

**Comment by brandonkelly** [comment, marked as answer]
This is pretty much resolved in Craft 5.10, where it’s now possible to replace the underlying custom field in a field layout element with a different field. (#18814)
